wom/src/test/scala/wdl4s/wdl/WdlWorkflowHttpImportSpec.scala

```diff
@@ -0,0 +1,108 @@
+package wdl4s.wdl
+
+import org.mockserver.integration.ClientAndServer
+import org.mockserver.integration.ClientAndServer.startClientAndServer
+import org.mockserver.model.HttpRequest.request
+import org.mockserver.model.HttpResponse.response
+import org.mockserver.socket.PortFactory
+import wdl.{ImportResolver, WdlNamespace, WdlNamespaceWithWorkflow}
+import org.scalatest.{FlatSpec, Matchers, BeforeAndAfterAll }
+
+class WdlWorkflowHttpImportSpec extends FlatSpec with BeforeAndAfterAll with Matchers  {
+  val tinyImport =
+    s"""
+       |task hello {
+       |  command {
+       |    echo "Hello World!"
+       |  }
+       |}
+       |
+       |workflow wf_hello {
+       |  call hello
+       |}
+       |
+     """.stripMargin
+
+  def tinyWorkflow(imp:String) =
+    s"""
+       | import "$imp" as imp
+       |
+       | workflow test_import {
+       |     call imp.hello
+       | }
+     """.stripMargin
+
+  val httpResolver: Seq[ImportResolver] = Seq(WdlNamespace.httpResolver)
+
+  var mockServer: ClientAndServer = _
+
+  override def beforeAll() = {
+    mockServer = startClientAndServer(PortFactory.findFreePort())
+
+    mockServer.when(
+      request().withPath("/hello.wdl")
+    ).respond(
+      response().withStatusCode(200).withBody(tinyImport)
+    )
+
+    mockServer.when(
+      request().withPath("/protected.wdl").withHeader("Authorization", "Bearer my-token-value")
+    ).
+      respond(
+      response().withStatusCode(200).withBody(tinyImport)
+    )
+
+    mockServer.when(
+      request().withPath("/redirect.wdl")
+    ).respond(
+      response().withStatusCode(301).withHeader("Location","/hello.wdl"))
+
+    mockServer.when(
+      request().withPath("/none.wdl")
+    ).respond(
+      response().withStatusCode(404)
+    )
+
+    () // explicitly return unit
```

Comment:
This might have been a(n unpleasant) surprise, but it's all over the code base and probably doesn't warrant a comment. 🙂

Reply:
Yeah, our linters are turned up to 11 :)